src/ansiblelint/rules/galaxy.py

```diff
@@ -103,10 +101,7 @@ def __eq__(self, other: object) -> bool:
 
     def __lt__(self, other: Version) -> bool:
         """Implement lower-than operation."""
-        try:
-            other = _coerce(other)
-        except NotImplementedError:
-            return NotImplemented
+        other = _coerce(other)
```

Comment:
Since the parameter is always going to be an instance of a Version class the `_coerce` function is not going to return an error. Hence removed the try except block